test/test_distributed.py

```diff
@@ -933,6 +979,7 @@ def _join_and_reduce(self, fn):
                     raise unittest.SkipTest("worldsize is too small to run group tests")
 
 elif BACKEND == 'mpi':
+    WORLD_SIZE = os.environ['WORLD_SIZE']
```

Comment:
Why is this necessary? Has it been failing before?

Reply:
yes, this variable will be used in the function _prepare_dummy_data(), and it should be initialized in the mpi backend, just like the other backend(https://github.com/pytorch/pytorch/blob/master/test/test_distributed.py#L853)